gov.pnnl.goss.gridappsd/applications/fncsgossbridge/apps/fncs_goss_bridge.py

```diff
@@ -60,63 +31,85 @@
-            logger.debug('Added isInitialized output, sending message '+str(message)+' connection '+str(gossConnection))
-
-            gossConnection.send(output_to_goss_topic , json.dumps(message))
-            gossConnection.send(output_to_goss_queue , json.dumps(message))
-        elif jsonmsg['command'] == 'update':
-            message['command'] = 'update'
-            _publishToFncsBus(simulationId, json.dumps(jsonmsg['message'])) #does not return
-        elif jsonmsg['command'] == 'nextTimeStep':
-            logger.debug('is next timestep')
-            message['command'] = 'nextTimeStep'
-            currentTime = jsonmsg['currentTime']
-            logger.debug('incrementing to '+str(currentTime))
-            _doneWithTimestep(currentTime) #currentTime is incrementing integer 0 ,1, 2.... representing seconds
-            logger.debug('done with timestep '+str(currentTime))
-            logger.debug('simulation id '+str(simulationId))
-            message['output'] = _getFncsBusMessages(simulationId)
-            responsemsg = json.dumps(message)
-            logger.debug('sending fncs output message '+str(responsemsg))
-
-            gossConnection.send(output_to_goss_topic , responsemsg)
-            gossConnection.send(output_to_goss_queue , responsemsg)
-        elif jsonmsg['command'] == 'stop':
+                responsemsg = json.dumps(message)
+                message_str = 'sending fncs output message '+str(responsemsg)
+                _sendSimulationStatus('running', message_str, 'debug')
+    
+                gossConnection.send(output_to_goss_topic , responsemsg)
+                gossConnection.send(output_to_goss_queue , responsemsg)
+            elif jsonmsg['command'] == 'stop':
+                message_str = 'Stopping the simulation'
+                _sendSimulationStatus('stopped', message_str, 'info')
+                fncs.die()
+                sys.exit()
+                
+        except Exception as e:
+            message_str = 'Error in command '+str(e)
+            _sendSimulationStatus('error', message_str, 'error')
+            if fncs.is_initialized():
+                fncs.die()
+           
+        
+    def on_error(self, headers, message):
+        message_str = 'Error in goss listener '+str(message)
+        _sendSimulationStatus('error', message_str, 'error')
+        if fncs.is_initialized():
             fncs.die()
-            sys.exit()
-    except Exception as e:
-        logger.error('Error in command '+str(e))
-  def on_error(self, headers, message):
-    logger.error('Error in goss listener '+str(message))
-  def on_disconnected(self):
-    logger.error('Disconnected')
+    
+    
+    def on_disconnected(self):
+        if fncs.is_initialized():
+            fncs.die()
+  
   
-def _registerWithFncsBroker(
-        simId, brokerLocation='tcp://localhost:5570'):
+def _registerWithFncsBroker(brokerLocation='tcp://localhost:5570'):
     '''Register with the fncs_broker and return.
```

Comment:
use """ instead of ''' for doc strings (this is my personal preference and my ide's personal preference).